pkg/resources/api_authentication_integration_with_authorization_code_grant.go

```diff
@@ -0,0 +1,523 @@
+					return diag.FromErr(err)
+				}
+			}
+			if v := d.GetRawConfig().AsValueMap()["oauth_client_auth_method"]; !v.IsNull() {
+				if err = d.Set("oauth_client_auth_method", v.AsString()); err != nil {
+					return diag.FromErr(err)
+				}
+			}
+			if v := d.GetRawConfig().AsValueMap()["oauth_authorization_endpoint"]; !v.IsNull() {
+				if err = d.Set("oauth_authorization_endpoint", v.AsString()); err != nil {
+					return diag.FromErr(err)
+				}
+			}
+			if v := d.GetRawConfig().AsValueMap()["oauth_token_endpoint"]; !v.IsNull() {
+				if err = d.Set("oauth_token_endpoint", v.AsString()); err != nil {
+					return diag.FromErr(err)
+				}
+			}
+			if v := d.GetRawConfig().AsValueMap()["oauth_allowed_scopes"]; !v.IsNull() {
+				if err = d.Set("oauth_allowed_scopes", ctyValToSliceString(v)); err != nil {
+					return diag.FromErr(err)
+				}
+			}
+			if v := d.GetRawConfig().AsValueMap()["oauth_grant"]; !v.IsNull() {
+				if err = d.Set("oauth_grant", v.AsString()); err != nil {
+					return diag.FromErr(err)
+				}
+			}
+			if v := d.GetRawConfig().AsValueMap()["comment"]; !v.IsNull() {
+				if err = d.Set("comment", v.AsString()); err != nil {
+					return diag.FromErr(err)
+				}
+			}
+		}
+
+		if err = d.Set(showOutputAttributeName, []map[string]any{schemas.SecurityIntegrationToSchema(integration)}); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err = d.Set(describeOutputAttributeName, []map[string]any{schemas.ApiAuthSecurityIntegrationPropertiesToSchema(properties)}); err != nil {
+			return diag.FromErr(err)
+		}
+
+		return nil
+	}
+}
+
+func UpdateContextApiAuthenticationIntegrationWithAuthorizationCodeGrant(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.AccountObjectIdentifier)
+	set, unset := sdk.NewApiAuthenticationWithAuthorizationCodeGrantFlowIntegrationSetRequest(), sdk.NewApiAuthenticationWithAuthorizationCodeGrantFlowIntegrationUnsetRequest()
+
+	if d.HasChange("comment") {
+		if v, ok := d.GetOk("comment"); ok {
+			set.WithComment(v.(string))
+		} else {
+			unset.WithComment(true)
+		}
+	}
+
+	if d.HasChange("enabled") {
+		if v := d.Get("comment").(string); v != "unknown" {
+			parsed, err := strconv.ParseBool(v)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			set.WithEnabled(parsed)
+		} else {
+			unset.WithEnabled(true)
+		}
+	}
+
+	if d.HasChange("oauth_access_token_validity") {
+		if v := d.Get("oauth_access_token_validity").(int); v != -1 {
+			set.WithOauthAccessTokenValidity(v)
+		} else {
+			// TODO: use UNSET
+			set.WithOauthAccessTokenValidity(0)
+		}
+	}
+
+	if d.HasChange("oauth_authorization_endpoint") {
+		set.WithOauthAuthorizationEndpoint(d.Get("oauth_authorization_endpoint").(string))
+	}
+
+	if d.HasChange("oauth_client_auth_method") {
+		v := d.Get("oauth_client_auth_method").(string)
+		if len(v) > 0 {
```

Comment:
and what else? if this one is supposed to be handled by forceNew, a comment would be helpful (also in other places)

Reply:
Added comments.